pkg/util/refreshable/refreshable.go

```diff
@@ -61,7 +61,7 @@ func (a *authorizer) RefreshWithContext(ctx context.Context, log *logrus.Entry)
 		if !autorest.IsTokenRefreshError(err) ||
 			azureerrors.IsUnauthorizedClientError(err) ||
 			azureerrors.IsInvalidSecretError(err) {
-			return false, nil
+			return false, err
```

Comment:
I think this was not printed out in the first place to not print out the client secret, It would probably be better to just create a custom error here to inform SRE about the client secret being invalid rather than using the azure error

Reply:
What ever be the error returned from here, the customer can view only the message and type what we set here in the "NewCloudError" function:
done, err = authorizer.RefreshWithContext(ctx, log)
		if err != nil {
			err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalCredentials, "properties.servicePrincipalProfile", "The provided service principal credentials are invalid.")
		}
		
My question is about what are the possible scenarios that we capture from those function in the if condition

Reply:
another way of handling this can be check whether done is false or not then check whether err is nil or not, then apply the custom message for false and nil values, in the parent function.

Reply:
Yep, I fully agree. Since this is only being used to call a generic error further upstream then this is okay.